src/main/java/no/sikt/nva/UnisContentScrapingResult.java

```diff
@@ -34,23 +38,36 @@ public static UnisContentScrapingResult fromWorkbook(Workbook workbook)
         validateSheet(sheet);
         validateHeaders(sheet.getRow(FIRST_ROW));
 
-        var results = new ArrayList<Record>();
+        var processedRecords = new HashMap<String, Record>();
         for (int i = SECOND_ROW; i <= getLastNonEmptyRowIndex(sheet); i++) {
             var unisContent = UnisContent.fromRow(sheet.getRow(i));
-            results.add(unisContent.toRecord());
+            processUnisContent(processedRecords, unisContent);
         }
 
         UnisContentScrapingResult result = new UnisContentScrapingResult();
-        result.setResults(results);
+        result.setResults(new ArrayList<>(processedRecords.values()));
         return result;
     }
 
+    public List<Record> getResults() {
+        return this.results;
+    }
+
     public void setResults(List<Record> results) {
         this.results = results;
     }
 
-    public List<Record> getResults() {
-        return this.results;
+    private static void processUnisContent(Map<String, Record> processedRecords, UnisContent unisContent)
+        throws URISyntaxException {
+
+        var newRecord = unisContent.toRecord();
+        if (processedRecords.containsKey(newRecord.getCristinId())) {
+            var existingRecord = processedRecords.get(newRecord.getCristinId());
+            var newContentFile = newRecord.getContentBundle().getContentFiles().get(0);
```

Comment:
I don't know anything about the unis stuff here, but is it always one and only one file here?

Reply:
Torbrjørn: NVA support multiple files per cristin-id. What Unis does is a bit unclear (we probably only need to ask), but a row in the excel sheet is a data related to a single file, and we don't know if multiple rows maps to the same cristin-id. So this code here is to catch that instance.

So what Joachim does here is correct.